File: clients/node/sdk/models/thumnailVersion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThumnailVersion = void 0;
class ThumnailVersion {
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    get additionalData() {
        return this._additionalData;
    }
    ;
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    set additionalData(value) {
        this._additionalData = value;
    }
    ;
    /**
     * Instantiates a new ThumnailVersion and sets the default values.
     */
    constructor() {
        this._additionalData = {};
    }
    ;
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    getFieldDeserializers() {
        return {
            "@1x": n => { this.onex = n.getStringValue(); },
            "@2x": n => { this.twox = n.getStringValue(); },
        };
    }
    ;
    /**
     * Gets the @1x property value. The Onex property
     * @returns a string
     */
    get onex() {
        return this._onex;
    }
    ;
    /**
     * Sets the @1x property value. The Onex property
     * @param value Value to set for the Onex property.
     */
    set onex(value) {
        this._onex = value;
    }
    ;
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    serialize(writer) {
        if (!writer)
            throw new Error("writer cannot be undefined");
        writer.writeStringValue("@1x", this.onex);
        writer.writeStringValue("@2x", this.twox);
        writer.writeAdditionalData(this.additionalData);
    }
    ;
    /**
     * Gets the @2x property value. The Twox property
     * @returns a string
     */
    get twox() {
        return this._twox;
    }
    ;
    /**
     * Sets the @2x property value. The Twox property
     * @param value Value to set for the Twox property.
     */
    set twox(value) {
        this._twox = value;
    }
    ;
}
exports.ThumnailVersion = ThumnailVersion;

File: clients/csharp/sdk/Models/ThumnailVersion.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System;
namespace MastodonClientLib.Models {
    public class ThumnailVersion : IParsable {
        /// <summary>The Onex property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Onex { get; set; }
#nullable restore
#else
        public string Onex { get; set; }
#endif
        /// <summary>The Twox property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Twox { get; set; }
#nullable restore
#else
        public string Twox { get; set; }
#endif
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static ThumnailVersion CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new ThumnailVersion();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"@1x", n => { Onex = n.GetStringValue(); } },
                {"@2x", n => { Twox = n.GetStringValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteStringValue("@1x", Onex);
            writer.WriteStringValue("@2x", Twox);
        }
    }
}
